tests/cpp_unit_tests/test_y_bus.cpp

```diff
@@ -272,6 +274,181 @@ TEST_CASE("Test fill-in y bus") {
+            CHECK(cabs(ybus.admittance()[i] - admittance_sym[i]) < numerical_tolerance);
+        }
+        ybus.update_admittance(std::make_shared<MathModelParam<true> const>(param_sym));
+
+        CHECK(ybus.admittance().size() == admittance_sym.size());
+        for (size_t i = 0; i < admittance_sym.size(); i++) {
+            CHECK(cabs(ybus.admittance()[i] - admittance_sym[i]) < numerical_tolerance);
+        }
+    }
+
+    SUBCASE("Test progressive update") {
+        YBus<true> ybus{topo_ptr, std::make_shared<MathModelParam<true> const>(param_sym)};
+        CHECK(ybus.admittance().size() == admittance_sym.size());
+        for (size_t i = 0; i < admittance_sym.size(); i++) {
+            CHECK(cabs(ybus.admittance()[i] - admittance_sym[i]) < numerical_tolerance);
+        }
+        auto branch_param_to_change_views =
+            std::views::iota(0, static_cast<int>(param_sym_update.branch_param.size())) |
+            std::views::filter([&param_sym_update](Idx i) {
+                return param_sym_update.branch_param[i].yff() != ComplexTensor<true>{0.0} ||
+                       param_sym_update.branch_param[i].yft() != ComplexTensor<true>{0.0} ||
+                       param_sym_update.branch_param[i].ytf() != ComplexTensor<true>{0.0} ||
+                       param_sym_update.branch_param[i].ytt() != ComplexTensor<true>{0.0};
+            });
+        auto shunt_param_to_change_views = std::views::iota(0, static_cast<int>(param_sym_update.shunt_param.size())) |
+                                           std::views::filter([&param_sym_update](Idx i) {
+                                               return param_sym_update.shunt_param[i] != ComplexTensor<true>{0.0};
+                                           });
+
+        MathModelParamIncrement<true> math_model_param_incrmt;
+        math_model_param_incrmt.branch_param = param_sym_update.branch_param;
+        math_model_param_incrmt.shunt_param = param_sym_update.shunt_param;
+        math_model_param_incrmt.source_param = param_sym_update.source_param; // not sure if we actually need this
+        math_model_param_incrmt.branch_param_to_change = {branch_param_to_change_views.begin(),
+                                                          branch_param_to_change_views.end()};
+        math_model_param_incrmt.shunt_param_to_change = {shunt_param_to_change_views.begin(),
+                                                         shunt_param_to_change_views.end()};
+
+        auto math_model_param_incrmt_ptr =
+            std::make_shared<MathModelParamIncrement<true> const>(math_model_param_incrmt);
+
+        ybus.update_admittance_increment(math_model_param_incrmt_ptr, false);
+        // check increment
+        CHECK(ybus.admittance().size() == admittance_sym_2.size());
+        for (size_t i = 0; i < admittance_sym_2.size(); i++) {
+            CHECK(cabs(ybus.admittance()[i] - admittance_sym_2[i]) < numerical_tolerance);
+        }
```

Comment:
since you use the same check so often, can you put it in a function in a nameless namespace in this file? e.g.

```cpp
namespace {
// function
} // namespace
```